### DISCUSSION TITLE
Request: Click on row, row details populate into popup modal

### DISCUSSION THREAD

**Discussion by danielrosehill** [Budibase/budibase#13975] [Help] [answered]
Anyone have a second to help me with this? Imagine it should be fairly straightforward.

Desired behavior (public element / frontend / world-facing):

-> Table populates
-> User clicks on row
-> Popup modal appears displaying a formatted view of the row contents

**Comment by mikesealey** [comment, marked as answer]
Hey @danielrosehill 

In your table you should set an _On row click_ action to _Navigate To_. For the time being, save this, and we'll come back to it.
https://github.com/Budibase/budibase/discussions/13796

Next, you'll want to create a new screen. In my example I'll be working with the sample data on the Employees table, when clicking a row, I want to see details about that employee. As such, my new screen will be called  `/employees/:id`. Notice the colon in the path there.

Go into the newly created screen and add a Data Provider component, set to fetch data from the same table. 
![image](https://github.com/Budibase/budibase/assets/39269380/c42e606e-7b98-4080-ba19-4cdff695249d)

Apply one filter on the `_id` column to match the binding of the ID passed in the URL (remember that colon?)
![image](https://github.com/Budibase/budibase/assets/39269380/9aaa2978-9adf-4dde-b425-fc01235a24e9)

Now, anything you add to your Data Provider component has access to the results of the rows that match the filter, which when filtering on _id is a single row - the row clicked in the original table. It's important to remember that the response of that Data Provider is an array though, a list of rows. 

You can now use the bindings to dig into what is inside that array of rows to populate your screen. I've used two bindings to display the first name, last name, and employee level
![image](https://github.com/Budibase/budibase/assets/39269380/4704a3f4-21b4-424d-b9d3-3b8622382274)

These values will change based on the values passed to the screen... Back to your original screen and it's Navigate To Action:
On Row Click triggers the action "Navigate To" which will open the page `/employees/:id` and pass the row id into the Data Provider and filter that we just set up. Checking "Open screen in modal" brings up a modal/popover card version of the screen we built.
![image](https://github.com/Budibase/budibase/assets/39269380/ce84d157-4484-48f7-8c4b-f98cae80c10f)
![image](https://github.com/Budibase/budibase/assets/39269380/a4a5c4db-dd60-471b-8f8b-d934ddfc3edb)

Make sure to remove the `/employees/:id` screen from the navigation, or it will be possible to navigate there without a `clicked-row _id`
![image](https://github.com/Budibase/budibase/assets/39269380/800c7be6-ca76-4201-bc72-5a89624fb1e1)

**Comment by danielrosehill** [comment]
![budibase](https://github.com/Budibase/budibase/assets/40233203/3ccd5b91-0407-446c-bc93-59428f12f53a)

@mikesealey Facing a very odd issue and just flagging in case it's a bug. The system won't let me create a screen with : on path. It automatically autoreplaces it to %3. Any idea?
